data_structures/stacks/stack_using_dll.py

```diff
@@ -0,0 +1,119 @@
+# A complete working Python program to demonstrate all  
+# stack operations using a doubly linked list  
+  
+# Node class  
+class Node: 
+  
+# Function to initialise the node object 
+    def __init__(self, data): 
+        self.data = data # Assign data 
+        self.next = None # Initialize next as null 
+        self.prev = None # Initialize prev as null         
+          
+# Stack class contains a Node object 
+class Stack: 

```

Comment:
Insert doctests here:
```python
    """
    >>> stack = Stack()
    >>> stack.isEmpty()
    True
    >>> for i in range(4):
    ...     stack.push(i)
    ...
    >>> stack.isEmpty()
    False
    >>> stack.printstack()
    stack elements are:
    3->2->1->0->
    >>> stack.top()
    3
    >>> stack.size()
    4
    >>> stack.pop()
    3
    >>> stack.printstack()
    stack elements are:
    2->1->0->
    """
```
And then locally do `python3 -m doctest stack_using_dll.py` and `python3 -m doctest -v stack_using_dll.py` and make sure those doctest pass.